crates/hir_ty/src/traits/chalk.rs

```diff
@@ -56,8 +56,13 @@ impl<'a> chalk_solve::RustIrDatabase<Interner> for ChalkContext<'a> {
     fn adt_datum(&self, struct_id: AdtId) -> Arc<StructDatum> {
         self.db.struct_datum(self.krate, struct_id)
     }
-    fn adt_repr(&self, _struct_id: AdtId) -> rust_ir::AdtRepr {
-        rust_ir::AdtRepr { repr_c: false, repr_packed: false }
+    fn adt_repr(&self, _struct_id: AdtId) -> Arc<rust_ir::AdtRepr<Interner>> {
+        // FIXME: keep track of these
+        Arc::new(rust_ir::AdtRepr { c: false, packed: false, int: None })
+    }
+    fn discriminant_type(&self, _ty: chalk_ir::Ty<Interner>) -> chalk_ir::Ty<Interner> {
+        // FIXME: keep track of this
+        unimplemented!();
```

Comment:
hmm this could potentiall introduce crashes, together with the `well_known_trait_from_lang_attr` addition. Could you make it instead just return `i32` (`chalk_ir::TyKind::Scalar(Scalar::Uint(chalk_ir::IntTy::I32)).intern(&Interner)`)?

Reply:
Updated, thanks

Reply:
What do you mean about `well_known_trait_from_lang_attr`?

Reply:
Just that because of that change, this function could be called.

Reply:
Ah, right.